Amend wording after discussion with Jonathan Wakely (#398)
jbcoe/value_types: DRAFT_COROLLARY.md

```

A. _Constraints_: `is_constructible_v<T, U>` is true.

>    `is_copy_constructible_v<U>` is `true`. `is_same_v<remove_cvref_t<U>,

Reviewer: ```suggestion
   `is_copy_constructible_v<T>` is `true`. `is_same_v<remove_cvref_t<U>,
```

E. _Mandates_: `T` is a complete type.

F. _Effects_: `alloc` is direct-non-list-initialized with `a`. Constructs an
    owned object of type `T` with `std​::​forward<U>(u)`, using the allocator
    `alloc`.

```c++
template<class I, class... Us>

> explicit constexpr indirect(in_place_t, std::initializer_list<I> ilist,

Reviewer: ```suggestion
explicit constexpr indirect(in_place_t, initializer_list<I> ilist,
```

Reviewer: and line 197 too

E. _Mandates_: `T` is a complete type.

F. _Effects_: `alloc` is direct-non-list-initialized with `a`.

>     Direct-non-list-initializes an owned object of type `T` with

Reviewer: ```suggestion
    Constructs an owned object of type `T` with
```

K. _Mandates_: `T` is a complete type.

L. _Effects_: `alloc` is direct-non-list-initialized with `a`.

>     Direct-non-list-initializes an owned object of type `T` with `ilist,

Reviewer: ```suggestion
    Constructs an owned object of type `T` with `ilist,
```

constexpr indirect& operator=(U&& u);
```

> A. _Constraints_: `is_constructible<T, U>` is true. `is_assignable<T,U>` is

Reviewer: ```suggestion
A. _Constraints_: `is_constructible_v<T, U>` is true. `is_assignable_v<T,U>` is
```

Author: Applied manually.

  true.

B. _Mandates_: `T` is a complete type.

> C. _Effects_: If `this` is valueless then equivalent to

Reviewer: ```suggestion
C. _Effects_: If `*this` is valueless then equivalent to
```

>    `*this = indirect<T>(std::forward<U>(u))`.

Reviewer: This uses a default constructed allocator, which is not listed as a constraint of calling this assignment operator. Should it be `*this = indirect(allocator_arg, alloc, std::forward<U>(u));` or should it be constrained to require `is_default_constructible_v<Allocator>`? Or something else?

Author: Nicely spotted. Agreed and applied manually.

D. _Constraints_: `is_base_of_v<T, std::remove_cvref_t<U>>` is `true`.
   `is_copy_constructible_v<remove_cvref_t<U>>` is `true`.

>    `is_same_v<remove_cvref_t<U>, in_place_t>` is `false`.

Reviewer: Should this have `remove_cvref_t<U>` is not a specialization of `in_place_type_t`? And similarly for the ctor above.

Author: `is_base_of_v<T, std::remove_cvref_t<U>>` is `true` should cover us as `in_place_type_t` is not derived from `T` or equal to `T` for any legal choice of `T`.

We can add the constraint if it helps to be explicit.

Author: Agree. `is_base_of_v<T, std::remove_cvref_t<U>>` is `true` should have use covered but no harm in being explicit - better errors that way.
